addons/l10n_ch/models/res_bank.py

```diff
@@ -332,14 +332,22 @@ def _is_qr_reference(self, reference):
                and re.match('\d+$', reference) \
                and reference == mod10r(reference[:-1])
 
-    def _eligible_for_qr_code(self, qr_method, debtor_partner, currency):
+    def _eligible_for_qr_code(self, qr_method, debtor_partner, currency, raises_error=True):
         if qr_method == 'ch_qr':
-
-            return self.acc_type == 'iban' and \
-                   self.partner_id.country_id.code == 'CH' and \
-                   (not debtor_partner or debtor_partner.country_id.code == 'CH') \
-                   and currency.name in ('EUR', 'CHF')
-
+            error_msg = "The QR code could not be generated for the following reason(s):"
+            if self.acc_type != 'iban':
+                error_msg += "The account type isn't QR-IBAN."
+            if self.partner_id.country_id.code != 'CH':
+                error_msg += "Your company isn't located in Switzerland."
+            if not debtor_partner or debtor_partner.country_id.code != 'CH':
+                error_msg += "The debtor partner's address isn't located in Switzerland."
+            if currency.id not in (self.env.ref('base.EUR').id, self.env.ref('base.CHF').id):
+                error_msg += "The currency isn't EUR nor CHF. \r\n"
+            if error_msg != "The QR code could not be generated for the following reason(s):":
```

Comment:
:scream: 

- the error message should be translated
- your sentence is wrongly build: "The account type isn't QR-IBAN.Your company isn't located in Switzerland." (need at least a space between all sentences)
- comparing string like that is a very bad design: you should append all messages into a list and then check if the list is empty or not instead